lldb/source/Plugins/SymbolFile/NativePDB/SymbolFileNativePDB.cpp

```diff
@@ -1630,6 +1630,53 @@ size_t SymbolFileNativePDB::ParseSymbolArrayInScope(
   return count;
 }
 
+void SymbolFileNativePDB::CacheTypeNames() {
```

Comment:
Actually, `SymbolFileNativePDB::BuildParentMap()` already does the tpi stream iteration and it's called at NativePDB plugin initial setup. We could just cache the those base names there instead of iterating it the second time.